### PR TITLE
docs: update `docs/usage/applications`

### PR DESCRIPTION
**Pull request by JacobCoffee**
## Description

- Updates `usage/applications` As part of #3212 work

### CODE REVIEW

File: docs/usage/applications.rst [hunk 1]
```diff
@@ -84,152 +85,139 @@ During shutdown, they are executed in the following order:
...
 
 Using Application State
 -----------------------
 
-As seen in the examples for the `on_startup / on_shutdown <Startup and Shutdown>`,
-callables passed to these hooks can receive an optional kwarg called ``app``, through which the application's state object
-and other properties can be accessed.
-The advantage of using application ``state``, is that it can be accessed during multiple stages of the connection, and
-it can be injected into dependencies and route handlers.
+As seen in the examples for the `on_startup / on_shutdown <Startup and Shutdown>`, :term:`callables <python:callable>`
```
Comment (JacobCoffee) [inline docs/usage/applications.rst:96]:
```suggestion
As seen in the examples for the `on_startup / on_shutdown <#startup-and-shutdown>`_, :term:`callables <python:callable>`
```

File: docs/usage/applications.rst [hunk 2]
```diff
@@ -84,152 +85,139 @@ During shutdown, they are executed in the following order:
...
-    :language: python
 
 .. _Initializing Application State:
 
 Initializing Application State
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-To seed application state, you can pass a :class:`State <.datastructures.state.State>` object to the ``state`` kwarg of
-the Litestar constructor:
+To seed application state, you can pass a :class:`.datastructures.state.State` object to the
```
Comment (JacobCoffee) [inline docs/usage/applications.rst:126]:
```suggestion
To seed application state, you can pass a :class:`~.datastructures.state.State` object to the
```

File: docs/usage/applications.rst [hunk 3]
```diff
@@ -238,20 +226,18 @@ Layered architecture
 
 Litestar has a layered architecture compromising of (generally speaking) 4 layers:
 
+#. The application object :class:`The application object <litestar.app.Litestar>`
+#. :class:`Routers <.router.Router>`
+#. `Controllers <.controller.Controller>`
+#. `Handlers <.handlers.BaseRouteHandler>`
```
Comment (JacobCoffee) [inline docs/usage/applications.rst:232]:
```suggestion
#. :class:`Controllers <.controller.Controller>`
#. :class:`Handlers <.handlers.BaseRouteHandler>`
```

**Review by cofin — APPROVED** [review]

File: docs/usage/applications.rst [hunk 4]
```diff
@@ -41,39 +44,37 @@ order, once the ASGI server (uvicorn, hypercorn, etc.) emits the respective even
 A classic use case for this is database connectivity. Often, we want to establish a database connection on application
 startup, and then close it gracefully upon shutdown.
 
-For example, lets create a database connection using the async engine from
+For example, let us create a database connection using the async engine from
 `SQLAlchemy <https://docs.sqlalchemy.org/en/latest/orm/extensions/asyncio.html>`_. We create two functions, one to get or
-establish the connection, and another to close it, and then pass them to the Litestar constructor:
+establish the connection, and another to close it, and then pass them to the :class:`~litestar.app.Litestar` constructor:
 
 .. literalinclude:: /examples/startup_and_shutdown.py
```
Comment (peterschutt) [inline docs/usage/applications.rst:51]: IDK why but this (and most of the other examples) on the page don't have full syntax highlighting in the docs preview. The Hello World example at the top does, but most others don't. E.g., keyword highlights, etc.

File: docs/usage/applications.rst [hunk 5]
```diff
@@ -84,152 +85,139 @@ During shutdown, they are executed in the following order:
...
-via the :class:`app.state <.app.Litestar>` attribute. As such it can be accessed wherever the app instance is accessible.
+:paramref:`~.app.Litestar.state` is one of the
+:ref:`reserved keyword arguments <usage/routing/handlers:"reserved" keyword arguments>`.
 
-It's important to understand in this context that the application instance is injected into the ASGI ``scope`` mapping for
-each connection (i.e. request or websocket connection) as ``scope["app"].state``. This makes the application accessible
-wherever the scope mapping is available, e.g. in middleware, on :class:`Request <.connection.request.Request>` and
-:class:`Websocket <.connection.websocket.WebSocket>` instances (accessible as ``request.app`` / ``socket.app``), and many
+It is important to understand in this context that the application instance is injected into the ASGI ``scope`` mapping
+for each connection (i.e. request or websocket connection) as ``scope["app"].state``. This makes the application
```
Comment (peterschutt) [inline docs/usage/applications.rst:110]:
```suggestion
for each connection (i.e. request or websocket connection) as ``scope["app"]``. This makes the application
```

File: docs/usage/applications.rst [hunk 6]
```diff
@@ -84,152 +85,139 @@ During shutdown, they are executed in the following order:
...
+To seed application state, you can pass a :class:`~.datastructures.state.State` object to the
+:paramref:`~.app.Litestar.state` parameter of the Litestar constructor:
 
 .. literalinclude:: /examples/application_state/passing_initial_state.py
     :caption: Using Application State
-    :language: python
 
+.. note:: :class:`~.datastructures.state.State` can be initialized with a :class:`dictionary <dict>`, an instance of
+    :class:`~.datastructures.state.ImmutableState` or :class:`~.datastructures.state.State`,
+    or a :class:`list` of :class:`tuples <tuple>` containing key/value pairs.
```
Comment (peterschutt) [inline docs/usage/applications.rst:134]: I don't think this is true anymore - maybe it was for 1.x but its just typed `State | None` now.

File: docs/usage/applications.rst [hunk 7]
```diff
@@ -84,152 +85,139 @@ During shutdown, they are executed in the following order:
...
+    ``True`` in the :class:`~.datastructures.state.State` constructor.
 
 Injecting Application State into Route Handlers and Dependencies
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 As seen in the above example, Litestar offers an easy way to inject state into route handlers and dependencies - simply
-by specifying ``state`` as a kwarg to the handler function. I.e., you can simply do this in handler function or dependency
-to access the application state:
+by specifying ``state`` as a kwarg to the handler function. In other words, you can simply do this in handler function
+or dependency to access the application state:
```
Comment (peterschutt) [inline docs/usage/applications.rst:145]:
```suggestion
by specifying ``state`` as a kwarg to the handler or dependency function. For example:
```
This was quite wordy, and felt to me like saying the same thing twice.

File: docs/usage/applications.rst [hunk 8]
```diff
@@ -84,152 +85,139 @@ During shutdown, they are executed in the following order:
...
-   @get("/")
-   def handler(state: State) -> None: ...
+    @get("/")
+    def handler(state: State) -> None: ...
 
 When using this pattern you can specify the class to use for the state object. This type is not merely for type
-checkers, rather Litestar will instantiate a new state instance based on the type you set there. This allows users to
-use custom classes for State, e.g.:
+checkers, rather Litestar will instantiate a new ``state`` instance based on the type you set there.
+This allows users to use custom classes for :class:`~.datastructures.state.State`, e.g.:
```
Comment (peterschutt) [inline docs/usage/applications.rst:159]:
```suggestion
This allows users to use custom classes for :class:`~.datastructures.state.State`.
```
No example follows.

File: docs/usage/applications.rst [hunk 9]
```diff
@@ -84,152 +85,139 @@ During shutdown, they are executed in the following order:
...
 
-While this is very powerful, it might encourage users to follow anti-patterns: it's important to emphasize that using
-state can lead to code that's hard to reason about and bugs that are difficult to understand, due to changes in
+While this is very powerful, it might encourage users to follow anti-patterns: it is important to emphasize that using
+state can lead to code that is hard to reason about and bugs that are difficult to understand, due to changes in
 different ASGI contexts. As such, this pattern should be used only when it is the best choice and in a limited fashion.
```
Comment (peterschutt) [inline docs/usage/applications.rst:163]: It feels weird to me that in an introductory usage document, we effectively say, "Hey, here's a thing you can do with `State`, but its weird and bug-prone, so don't do it."

Makes me feel that we should have a separate section in usage for `State` that explores these concepts outside of the app use docs. WDYT?

File: docs/usage/applications.rst [hunk 10]
```diff
@@ -84,152 +85,139 @@ During shutdown, they are executed in the following order:
...
 
-.. note::
-
-    ``on_app_init`` handlers cannot be ``async def`` functions, as they are called within ``Litestar.__init__()``,
-    outside of an async context.
+.. note:: :paramref:`~.app.Litestar.on_app_init` handlers cannot be :ref:`python:async def` functions, as they are
+    called within :paramref:`~litestar.app.Litestar.__init__`, outside of an async context.
 
 .. literalinclude:: /examples/application_hooks/on_app_init.py
     :caption: After Exception Hook
```
Comment (peterschutt) [inline docs/usage/applications.rst:220]: Incorrect caption

File: docs/usage/applications.rst [hunk 11]
```diff
@@ -238,20 +226,18 @@ Layered architecture
 
 Litestar has a layered architecture compromising of (generally speaking) 4 layers:
```
Comment (peterschutt) [inline docs/usage/applications.rst:227]:
```suggestion
Litestar has a layered architecture compromising of 4 layers:
```

File: docs/usage/applications.rst [hunk 12]
```diff
@@ -238,20 +226,18 @@ Layered architecture
...
 
+#. The application object :class:`The application object <litestar.app.Litestar>`
```
Comment (peterschutt) [inline docs/usage/applications.rst:229]:
```suggestion
#. :class:`The application object <litestar.app.Litestar>`
```

**Review by peterschutt — APPROVED** [review]

File: docs/usage/applications.rst (hunk 6 above)
Comment (JacobCoffee) [reply docs/usage/applications.rst]: wdyt https://github.com/litestar-org/litestar/commit/1e5df3296abd3acd452d6e51d3006f98ac824146

File: docs/usage/applications.rst (hunk 4 above)
Comment (JacobCoffee) [reply docs/usage/applications.rst:51]: Im not seeing why it does this either :|

Comment (JacobCoffee) [reply docs/usage/applications.rst:51]: tried tagging it with `:language:`, matching what the working one does, nothing.. not sure if its a local thing?

File: docs/usage/applications.rst (hunk 10 above)
Comment (JacobCoffee) [reply docs/usage/applications.rst]: fixed in 689d47537bb368d6a660a6673647f00b92528ddf

File: docs/usage/applications.rst (hunk 9 above)
Comment (JacobCoffee) [reply docs/usage/applications.rst]: What's a good spot you think? a topic? or a section in this further down maybe under `Advanced Stuff`?

**Comment by JacobCoffee** [comment]
Should be reopened and pointed to `main`

File: docs/usage/applications.rst (hunk 6 above)
Comment (peterschutt) [reply docs/usage/applications.rst]: Sorry - I was wrong about this. I thought this was referring to the value that could be passed to the application on instantiation, not what can be passed to the state object on instantiation.

File: docs/usage/applications.rst (hunk 9 above)
Comment (peterschutt) [reply docs/usage/applications.rst]: Opened #3346 for this

**Pull request closed without merging**